azure/interfaces.go

```diff
@@ -138,14 +138,14 @@ type ResourceSpecGetterWithHeaders interface {
 }
 
 // ASOResourceSpecGetter is an interface for getting all the required information to create/update/delete an Azure resource.
-type ASOResourceSpecGetter interface {
+type ASOResourceSpecGetter[T genruntime.MetaObject] interface {
```

Comment:
`T` here ultimately maps to an ASO root object type represented in Kubernetes like `ResourceGroup` or `VirtualMachine`.

The way I reason about this is that each `ASOResourceSpecGetter` is parameterized over a particular type, say `ResourceGroup`. That means a `ASOResourceSpecGetter[*ResourceGroup]` requires a method with the signature
```
ResourceRef() *ResourceGroup
```
Whereas before an implementation could return any type that implements the `genruntime.MetaObject` interface, where there's a risk that a non-ResourceGroup could be returned and lead to unexpected errors.